### ISSUE TITLE
Accessibility support

### ISSUE THREAD

**Issue by Saulis** [vaadin/vaadin-button#6]
User should be able to define a label which gets announced by a screen reader when the button gets focused.

Also, correct aria roles and states should be set, see: https://www.w3.org/TR/wai-aria-practices-1.1/#button

- reader should announce contents by default, and `aria-label` can be used to override the announcement
